dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/GridUtils.java

```diff
@@ -293,6 +294,23 @@ public static void toXls(Grid grid, OutputStream out) throws IOException {
     workbook.close();
   }
 
+  /** Writes a XLSX (Excel workbook) representation of the given Grid to the given OutputStream. */
+  public static void toXlsx(Grid grid, OutputStream out) throws IOException {
+    Workbook workbook = new XSSFWorkbook();
+
+    String sheetName =
```

Comment:
The lines 301 to 311 are same as lines 284 to 294.
I suggest we extract this common code into a separate private method and reuse this new method on both places.